### PR TITLE
[KeyVault] Support for API version 7.5-preview.1

### PR DESCRIPTION
**Pull request by gracewilcox**
Regenerated code with the new 7.5-preview.1 swagger and rerecorded tests.

Features added:
* The `HsmPlatform` field to `KeyAttributes`
* Managed Identity auth support for backup and restore operations

Closes https://github.com/Azure/azure-sdk-for-go/issues/21561

### CODE REVIEW

File: sdk/security/keyvault/azadmin/backup/autorest.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ clear-output-folder: false
 export-clients: true
 go: true
 input-file: 
-    - https://github.com/Azure/azure-rest-api-specs/blob/551275acb80e1f8b39036b79dfc35a8f63b601a7/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.4/backuprestore.json
+    - https://github.com/Azure/azure-rest-api-specs/blob/5dd1107d5f2be8d600325d795450e1d854fbe7e8/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.5-preview.1/backuprestore.json
```
Comment (heaths) [inline sdk/security/keyvault/azadmin/backup/autorest.md:8]: Should be commit a2f6f742d088dcc712e67cb2745d8271eaa370ff. Not sure where this one came from.

File: sdk/security/keyvault/azkeys/client_test.go [hunk 2]
```diff
@@ -537,69 +537,63 @@ func TestRecoverDeletedKey(t *testing.T) {
...
-					},
-				}
-				createResp, err = client.CreateKey(context.Background(), key, params, nil)
-				if err == nil {
-					break
-				}
-				if i < 4 {
-					recording.Sleep(30 * time.Second)
-				}
+	t.Run("KV", func(t *testing.T) {
```
Comment (heaths) [inline sdk/security/keyvault/azkeys/client_test.go:540]: So you're not going to test MHSM in the future? Why not just comment out the condition for now?

File: sdk/security/keyvault/azkeys/go.mod [hunk 3]
```diff
@@ -5,7 +5,7 @@ go 1.18
 require (
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.9.0-beta.1
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.3.1
-	github.com/Azure/azure-sdk-for-go/sdk/internal v1.4.0
+	github.com/Azure/azure-sdk-for-go/sdk/internal v1.3.0
```
Comment (heaths) [inline sdk/security/keyvault/azkeys/go.mod:8]: Wait...just noticed this is a downgrade. Is that expected?

**Review by heaths — CHANGES_REQUESTED** [review]

File: sdk/security/keyvault/azkeys/go.mod (hunk 3 above)
Comment (gracewilcox) [reply sdk/security/keyvault/azkeys/go.mod:8]: Having issues with the newer version of sdk/internal. Its recording package is breaking my test proxy and causing the pipelines to fail. Working fixing it now

File: sdk/security/keyvault/azadmin/backup/autorest.md (hunk 1 above)
Comment (gracewilcox) [reply sdk/security/keyvault/azadmin/backup/autorest.md:8]: Commit I'm using came from the issue Pallavi made, changing to the new commit

File: sdk/security/keyvault/azkeys/client_test.go (hunk 2 above)
Comment (gracewilcox) [reply sdk/security/keyvault/azkeys/client_test.go]: Changed it so MHSM are just commented out for now.

File: sdk/security/keyvault/azadmin/backup/autorest.md (hunk 1 above)
Comment (gracewilcox) [reply sdk/security/keyvault/azadmin/backup/autorest.md]: fixed in [9afbf80](https://github.com/Azure/azure-sdk-for-go/pull/21910/commits/9afbf80857d5a559ad6c24a6fddf24714a2a3ee5)

File: sdk/security/keyvault/azkeys/go.mod (hunk 3 above)
Comment (gracewilcox) [reply sdk/security/keyvault/azkeys/go.mod]: fixed in [f1a71c1](https://github.com/Azure/azure-sdk-for-go/pull/21910/commits/f1a71c116e1b8fd1bea0a1836a658ec274a90756)

File: sdk/security/keyvault/azkeys/client_test.go (hunk 2 above)
Comment (gracewilcox) [reply sdk/security/keyvault/azkeys/client_test.go]: fixed in [d6507a9](https://github.com/Azure/azure-sdk-for-go/pull/21910/commits/d6507a98f031e9f4bcfcbb9a86b87b108186cbae)

**Review by heaths — APPROVED** [review]

**Pull request merged**